internal/kafka/internal/workers/cluster_mgrs/cleanup_clusters_mgr.go

```diff
@@ -0,0 +1,113 @@
+package cluster_mgrs
+
+import (
+	fleeterrors "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	cleanupClustersWorkerType = "cleanup_clusters"
+)
+
+// CleanupClustersManager represents a worker that periodically reconciles data plane clusters in cleanup state.
+type CleanupClustersManager struct {
+	workers.BaseWorker
+	clusterService             services.ClusterService
+	kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon
+	osdIdpKeycloakService      sso.OsdKeycloakService
```

Comment:
let's call this `osdIDPKeycloakService`: acronyms in Go should be written in uppercase and not mixed case (as long as it doesn't affect exportability), see https://github.com/golang/go/wiki/CodeReviewComments#initialisms